pennylane/fourier/reconstruct.py

```diff
@@ -0,0 +1,640 @@
+            of an argument of ``qnode`` .
+            If a ``dict`` , the values of ``ids`` have to contain the parameter indices
+            for the respective array-valued QNode argument represented by the key.
+            These indices always are tuples, i.e. ``()`` for scalar and ``(i,)`` for
+            one-dimensional arguments.
+            If a ``list`` , the parameter indices are inferred from ``nums_frequency`` if
+            given or ``spectra`` else.
+            If ``None``, all keys present in ``nums_frequency`` / ``spectra`` are considered.
+        nums_frequency (dict[dict]): Numbers of integer frequencies -- and biggest
+            frequency -- per QNode parameter. The keys have to be argument names of ``qnode``
+            and the inner dictionaries have to be mappings from parameter indices to the
+            respective integer number of frequencies. If the QNode frequencies are not contiguous
+            integers, the argument ``spectra`` should be used to save evaluations of ``qnode`` .
+            Takes precedence over ``spectra`` and leads to usage of equidistant shifts.
+        spectra (dict[dict]): Frequency spectra per QNode parameter.
+            The keys have to be argument names of ``qnode`` and the inner dictionaries have to
+            be mappings from parameter indices to the respective frequency spectrum for that
+            parameter. Ignored if ``nums_frequency!=None``.
+        shifts (dict[dict]): Shift angles for the reconstruction per QNode parameter.
+            The keys have to be argument names of ``qnode`` and the inner dictionaries have to
+            be mappings from parameter indices to the respective shift angles to be used for that
+            parameter. For :math:`R` non-zero frequencies, there must be :math:`2R+1` shifts
+            given. Ignored if ``nums_frequency!=None``.
+        atol (float): Absolute tolerance used to analyze shifts lying close to 0.
+
+    Returns:
+        dict[dict]: Indices for the QNode parameters with respect to which the QNode
+            will be reconstructed. Cast to the dictionary structure explained above.
+            If the input ``ids`` was a dictionary, it is returned unmodified.
+        callable: The reconstruction method to use, one out of two internal methods.
+        dict[dict[dict]]: Keyword arguments for the reconstruction method specifying
+            how to carry out the reconstruction. The outer-most keys are QNode argument
+            names, the middle keys are parameter indices like the inner keys of
+            ``nums_frequency`` or ``spectra`` and the inner-most dictionary contains the
+            keyword arguments, i.e. the keys are keyword argument names for the
+            reconstruction method
+        bool: Whether any of the reconstruction jobs will require the evaluation
+            of the function at the position of reconstruction itself.
+    """
+    # pylint: disable=too-many-branches
```

Comment:
😆 I think this is pylint trying to tell you to break this up into multiple functions (I have no great opinion on this, happy with it as is)